### PR TITLE
Bug fixed on mismatched skey & remove restriction on min amount

### PR DESCRIPTION
**Pull request by saifiikmal**
Hi @andrewpaliga ,

I think this bugs has to do with the errors on the shopify offsite_gateway_callback

### CODE REVIEW

**Comment by andrewpaliga** [comment]
LGTM, @girasquid?

File: lib/offsite_payments/integrations/molpay.rb [hunk 1]
```diff
@@ -60,8 +60,7 @@ def form_fields
         end
 
         def amount=(money)
-          #Molpay minimum amount is 1.01
-          if money.is_a?(String) or money.to_f < 1.01
+          if money.is_a?(String)
```
Comment (camilo) [inline lib/offsite_payments/integrations/molpay.rb]: why not test for what we want ? as in `unless money.is_a?(Money) || money <= 0 `

Comment (saifiikmal) [inline lib/offsite_payments/integrations/molpay.rb]: ok will change this. Any other things? @camilo @andrewpaliga

Comment (andrewpaliga) [inline lib/offsite_payments/integrations/molpay.rb]: @saifiikmal I believe the suggestion was to be more explicit and use `unless money.is_a?(Money) || money <= 0` vs. `if money.is_a?(String)`

**Comment by saifiikmal** [comment]
@andrewpaliga i've made the changes..can u have a look?

**Comment by andrewpaliga** [comment]
Why not test for what you're looking for like: `unless money.is_a?(Money) && money >= 0'?

**Comment by andrewpaliga** [comment]
Either that, or remove the check for datatype all together as we don't use it for universal: https://github.com/Shopify/offsite_payments/blob/master/lib/offsite_payments/integrations/universal.rb#L72

**Comment by saifiikmal** [comment]
You mean like this ```unless money > 0``` ?

**Comment by andrewpaliga** [comment]
Not sure if you want to reject $0 orders, but if so, then yes.

**Comment by saifiikmal** [comment]
ok will make changes accordingly..tq

**Comment by andrewpaliga** [comment]
:+1:

**Comment by girasquid** [comment]
:+1:

**Pull request merged**